src/utils/classifier_report.py

```diff
@@ -0,0 +1,108 @@
+
+                # Datables CDN
+                a.script(src="https://cdn.datatables.net/1.13.4/js/jquery.dataTables.js")
+                a.script(src="https://cdn.datatables.net/searchpanes/2.1.2/js/dataTables.searchPanes.min.js")
+                a.script(src="https://cdn.datatables.net/select/1.6.2/js/dataTables.select.min.js")
+                a.script(src="https://cdn.datatables.net/fixedheader/3.3.2/js/dataTables.fixedHeader.min.js")
+                a.script(src="https://cdn.datatables.net/responsive/2.4.1/js/dataTables.responsive.min.js")
+                
+                # Javascript creating the table
+                #  the 'Path' column accounts for the csv 
+                with a.script():
+                    a(f'''
+                        let table = new DataTable('#myTable', {{
+                            data: {self.data},
+                            responsive: true,
+                            fixedHeader:{{ header: true, footer: true }},
+                            ordering: true,
+                            paging: true,
+                            scrollY: true,
+                            searching: true,
+                            searchPanes: {{viewTotal: true, layout: 'columns-4', initCollapsed: true}},
+                            dom: 'Plfrtip',
+                            columns: [
+                                {{ title: 'Image', data: 'normalized_path', render: (d,t,r,m) => '<img src="'+d+'" style=height:200px; />'}},
+                                {{ title: 'Path', data: 'original_path'}},
+                                {{ title: 'Class', data: 'predicted_class'}},
+                                {{ title: 'Confidence', data: 'predicted_conf'}}
+                            ],
+                        }});
+                    ''')
+        
+        self.html_page = a
+    
+    # save file in predefined reports folder or into the defined directory
+    def save_file(self, output_path):
+       
+        html = str(self.html_page)
+        
+        # if save path was not specified, creates reports directory and reports.html file within it
+        if output_path:
+             with open(output_path, 'w') as f:
+                f.write(html)
+        else:
+            dirname = os.getcwd()
+            report_dir = os.path.join(dirname, 'reports/')
+            
+            if not os.path.exists(report_dir):
+                os.mkdir(report_dir)
+                print('directory has been created')
+            
+            output_path = os.path.join(report_dir, "report.html")
+            
+            with open(output_path, 'w') as f:
+                f.write(html)            
+        
+        print(f"HTML file saved at {output_path}")
+        self.url = f"file://{os.path.abspath(output_path)}"
+
+
+
+    # launches the HTML page in default browser
+    def launch_page(self):
+        webbrowser.open_new_tab(self.url)
+
+
+ReportGenerator()
```

Comment:
I don't think we need in instantiate the class here anymore, since it's getting instantiated in create_report.py